lib/matplotlib/colors.py

```diff
@@ -286,6 +286,11 @@ def to_rgba(c, alpha=None):
         Tuple of floats ``(r, g, b, a)``, where each channel (red, green, blue,
         alpha) can assume values between 0 and 1.
     """
+    if isinstance(c, tuple) and len(c) == 2 and isinstance(c[0], str):
+        if 0 <= c[1] <= 1:
+            c, alpha = c[0], c[1]
+        else:
+            raise ValueError('Alpha must be between 0 and 1, inclusive.')
```

Comment:
Afaict this check is redundant with the one in _to_rgba_no_colorcycle and should be removed.

Reply:
There is likely a better place for this check, but I'm not sure it is redundant. If we remove this check and keep `c, alpha = c[0], c[1]` there, calling `to_rgba(('blue', 2))` calls `_to_rgba_no_colorcycle('blue', 2)` and returns `(0.0, 0.0, 1.0, 2)` from [line 343](https://github.com/j1642/matplotlib/blob/9b308458cf592f08ee23f1d2c547934b7d15b0c3/lib/matplotlib/colors.py#L343) with an invalid alpha value.

Reply:
Ah, good catch, I missed the early returns in _to_rgba_no_colorcycle.  Still, one would cover more cases (e.g. `to_rgba("blue", 2)`) by instead checking for `if alpha is not None and not 0 <= alpha <= 1` at (or close to) the top of _to_rgba_no_colorcycle.

Reply:
+1 for deferring the alpha limits check to `to_rgba_no_color_cylce`.

Additionally, we want to keep 

> If *alpha* is given, force the alpha value of the returned RGBA tuple
to *alpha*.

You cannot reassign alpha unconditionally. I think you need to
```
if alpha is None:
    c, alpha = c
else:
    c = c[0]  # discard alpha from tuple
```